Src/HALALMock/Services/Communication/Ethernet/TCP/Socket.cpp

```diff
@@ -1,245 +1,283 @@
-	connection_callback();
+void Socket::connect_attempt() {
+    // insert the remote address and port and connect
+    struct sockaddr_in remote_addr;
+    remote_addr.sin_family = AF_INET;
+    remote_addr.sin_addr.s_addr = remote_ip.address;
+    remote_addr.sin_port = htons(remote_port);
+    if (connect(socket_fd, (struct sockaddr*)&remote_addr,
+                sizeof(remote_addr)) < 0) {
+        state = INACTIVE;
+        return;
+    }
+    LOG_INFO("Connection established with the remote socket");
+    connection_callback();
 }
 
-void Socket::connection_callback(){
-	EthernetNode remote_node(remote_ip, remote_port);
-	if(connecting_sockets.contains(remote_node)){
-		connecting_sockets.erase(remote_node);
-		state = CONNECTED;
-	}
-	//start receiving
-	is_receiving = true;
-    receiving_thread = std::jthread(&Socket::receive, this); 
+void Socket::connection_callback() {
+    EthernetNode remote_node(remote_ip, remote_port);
+    if (connecting_sockets.contains(remote_node)) {
+        connecting_sockets.erase(remote_node);
+        state = CONNECTED;
+    }
+    // start receiving
+    is_receiving = true;
+    receiving_thread = std::jthread(&Socket::receive, this);
 }
 
-Socket::Socket(IPV4 local_ip, uint32_t local_port, IPV4 remote_ip, uint32_t remote_port, uint32_t inactivity_time_until_keepalive, uint32_t space_between_tries, uint32_t tries_until_disconnection): Socket(local_ip, local_port, remote_ip, remote_port){
-	keepalive_config.inactivity_time_until_keepalive = inactivity_time_until_keepalive;
-	keepalive_config.space_between_tries = space_between_tries;
-	keepalive_config.tries_until_disconnection = tries_until_disconnection;
+Socket::Socket(IPV4 local_ip, uint32_t local_port, IPV4 remote_ip,
+               uint32_t remote_port, uint32_t inactivity_time_until_keepalive,
+               uint32_t space_between_tries, uint32_t tries_until_disconnection)
+    : Socket(local_ip, local_port, remote_ip, remote_port) {
+    keepalive_config.inactivity_time_until_keepalive =
+        inactivity_time_until_keepalive;
+    keepalive_config.space_between_tries = space_between_tries;
+    keepalive_config.tries_until_disconnection = tries_until_disconnection;
 }
 
-Socket::Socket(EthernetNode local_node, EthernetNode remote_node):Socket(local_node.ip, local_node.port, remote_node.ip, remote_node.port){}
+Socket::Socket(EthernetNode local_node, EthernetNode remote_node)
+    : Socket(local_node.ip, local_node.port, remote_node.ip, remote_node.port) {
+}
 
-void Socket::close(){
-	::close(socket_fd);
+void Socket::close() {
+    ::close(socket_fd);
     if (is_receiving) {
         is_receiving = false;
         if (receiving_thread.joinable()) {
-            receiving_thread.join();  
+            receiving_thread.join();
         }
     }
-	while(!tx_packet_buffer.empty()){
-		tx_packet_buffer.pop();
-	}
-	state = CLOSING;
+    while (!tx_packet_buffer.empty()) {
+        tx_packet_buffer.pop();
+    }
+    state = CLOSING;
```

Comment:
could be a good idea add a LOG_INFO when the socket is closed
```suggestion
    state = CLOSING;
    LOG_INFO("Socket has been closed correctly")
```